compiler/rustc_trait_selection/src/solve/alias_relate.rs

```diff
@@ -84,3 +93,105 @@ impl<'tcx> EvalCtxt<'_, InferCtxt<'tcx>> {
         }
     }
 }
+
+enum IgnoreAliases {
+    Yes,
+    No,
+}
+
+impl<'tcx> EvalCtxt<'_, InferCtxt<'tcx>> {
+    /// In case a rigid term refers to a placeholder which is not referenced by the
+    /// alias, the alias cannot be normalized to that rigid term unless it contains
+    /// either inference variables or these placeholders are referenced in a term
+    /// of a `Projection`-clause in the environment.
+    fn alias_cannot_name_placeholder_in_rigid(
+        &mut self,
+        param_env: ty::ParamEnv<'tcx>,
+        rigid_term: ty::Term<'tcx>,
+        alias: ty::Term<'tcx>,
+    ) -> bool {
+        // Check that the rigid term is actually rigid.
+        if rigid_term.to_alias_term().is_some() || alias.to_alias_term().is_none() {
+            return false;
+        }
+
+        // If the alias has any type or const inference variables,
+        // do not try to apply the fast path as these inference variables
+        // may resolve to something containing placeholders.
+        if alias.has_non_region_infer() {
+            return false;
+        }
+
+        let mut referenced_placeholders =
```

Comment:
Why `IgnoreAliases::Yes`?

Reply:
Feels like this fast reject could be quickly side-stepped w/ something like `Mirror<!A> alias-relate Mirror<!B>`

Reply:
Update: It occurred to me last night while I was sleeping that we ignore aliases here b/c e.g. `Alias<!A>` could normalize to sth not mentioning `!A`.